src/Purestone/Server/App.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Purestone.Server.App (app) where
+
+import Servant.Server
+import Servant
+import Data.IORef
+import Data.Time.Clock
+
+import Purestone.Board
+import Purestone.Card
+import Purestone.Deck
+import Purestone.Server.GetState
+import Purestone.Server.MakeMove
+import Purestone.Server.GameState
+import Purestone.Server.ConnectResponse
+import Purestone.Server.Util
+
+type PurestoneAPI =  "getState"  :> Capture "gameID" Int :> Capture "player" Int :> QueryParam "Last-Received-Update" UTCTime :> Get '[JSON] Board
```

Comment:
That seems like a pretty long name for the parameter, how about just "update"? The path for this would currently be something like `/getState/0/1/?Last-Received-Updated=something`

Reply:
I think potentially, it would also make sense to use `state` instead of `getState` since the "get" part is handled by the `GET` verb? Then later the same path could be used for updating the game state with e.g. `POST /state/...` or similar.

Reply:
> That seems like a pretty long name for the parameter [...]

In principle this doesn't need to be a query parameter at all. If it's provided as a header per the specification of [`if-modified-since`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since), it can be omitted from the URI.

> I think potentially, it would also make sense to use `state` [...]

Yeah, this is something I thought of as well. `GET /state` is nice. However you wouldn't want to use `POST /state`, since the thing being POSTed is a move rather than the gamestate itself. `POST /move` would work though.

Reply:
> In principle this doesn't need to be a query parameter at all. If it's provided as a header per the specification of [`if-modified-since`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since), it can be omitted from the URI.

We discussed this on the Slack yesterday and decided against using the header.

> Yeah, this is something I thought of as well. `GET /state` is nice. However you wouldn't want to use `POST /state`, since the thing being POSTed is a move rather than the gamestate itself. `POST /move` would work though.

I don't think that's truly REST-ful design. I think the most appropriate choice would probably be something like `PATCH /state/...` since the state of a particular game is being modified. If you wanted to have a separate endpoint for moves, then each move would have to be given a unique identifier and you'd have to go with something like `PUT /move` and it would then be a bit odd for it to have modified the response you get from `GET /state/`

Reply:
> We discussed this on the Slack yesterday and decided against using the header.

I can't see a conversation on this, can you link it? (or maybe outline the reasons here for posterity?)

> I think the most appropriate choice would probably be something like `PATCH /state/...` since the state of a particular game is being modified.

My understanding was that when using PATCH you specify the changes to be made explicitly, whereas in this case we simply tell the server what we've done, and the server figures out what that means for the state change. I agree that having the endpoints for modifying and retrieving the state be different is a little un-RESTful, but having the server dictate the precise state transformation is un-RESTful already so there is no good idiom for it.

Reply:
I've implemented those changes, let me know if the comments need improving. I'm just going to also put `setupGame` and `processMove` into a `Purestone.Game` module (or `Purestone.Play`?) because it feels wrong to keep temporary definitions in the server files for later removal

Reply:
@dixonary It was actually in a private chat, the reasoning was mainly due to that HTTP header using a strange date format, whereas with our own query parameter we could use standard `UTCTime` functions in both the server and client without having to define custom formatting strings to work with it

Reply:
> @dixonary It was actually in a private chat, the reasoning was mainly due to that HTTP header using a strange date format [...]

Reasonable enough! It is annoying that they don't use a standardized time format.

>  I'm just going to also put setupGame and processMove into a Purestone.Game module

That's a good idea. It might be convenient for the client to be able to predict what will happen when a move is made and to be able to check its legality anyway, so having those functions more widely available is handy.